```
{Failed;Failed;Failed;Failed;  === RUN   Test_UncertainDeviceGlobalMounts/timed_out_operations_should_result_in_device_marked_as_uncertain_[Filesystem][
=== PAUSE Test_UncertainDeviceGlobalMounts/timed_out_operations_should_result_in_device_marked_as_uncertain_[Filesystem][
=== CONT  Test_UncertainDeviceGlobalMounts/timed_out_operations_should_result_in_device_marked_as_uncertain_[Filesystem][
    reconciler_test.go:1654: Error verifying UnMountDeviceCallCount: Expected DeviceUnmount Call 1, got 0
--- FAIL: Test_UncertainDeviceGlobalMounts/timed_out_operations_should_result_in_device_marked_as_uncertain_[Filesystem][ (0.34s)
;=== RUN   Test_UncertainDeviceGlobalMounts
--- FAIL: Test_UncertainDeviceGlobalMounts (0.00s)
;=== RUN   Test_UncertainVolumeMountState/failed_operation_should_result_in_not-mounted_volume_[Filesystem]
=== PAUSE Test_UncertainVolumeMountState/failed_operation_should_result_in_not-mounted_volume_[Filesystem]
=== CONT  Test_UncertainVolumeMountState/failed_operation_should_result_in_not-mounted_volume_[Filesystem]
    reconciler_test.go:1873: Error verifying UnMountDeviceCallCount: Expected DeviceUnmount Call 1, got 0
--- FAIL: Test_UncertainVolumeMountState/failed_operation_should_result_in_not-mounted_volume_[Filesystem] (1.07s)
;=== RUN   Test_UncertainVolumeMountState
--- FAIL: Test_UncertainVolumeMountState (0.00s)
;}
```

The new failure seems unrelated: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128387/pull-kubernetes-unit/1850925531751518208

/test pull-kubernetes-unit

--- a/pkg/controller/job/job_controller_test.go
+++ b/pkg/controller/job/job_controller_test.go
@@ -38,6 +38,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/rand"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/uuid"
+	utilversion "k8s.io/apimachinery/pkg/util/version"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/apiserver/pkg/util/feature"
@@ -1249,6 +1250,10 @@ func TestControllerSyncJob(t *testing.T) {
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
 			logger, _ := ktesting.NewTestContext(t)
+			if tc.podIndexLabelDisabled {
+				// TODO: this will be removed in 1.35 when 1.31 will fall out of support matrix
+				featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, feature.DefaultFeatureGate, utilversion.MustParse("1.31"))
+			}
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodIndexLabel, !tc.podIndexLabelDisabled)
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobPodReplacementPolicy, tc.jobPodReplacementPolicy)
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobSuccessPolicy, tc.jobSuccessPolicy)

--- a/pkg/controller/statefulset/stateful_set_control_test.go
+++ b/pkg/controller/statefulset/stateful_set_control_test.go
@@ -39,6 +39,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	utilversion "k8s.io/apimachinery/pkg/util/version"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/client-go/informers"
 	appsinformers "k8s.io/client-go/informers/apps/v1"
@@ -165,7 +166,8 @@ func TestStatefulSetControl(t *testing.T) {
 		fn  func(*testing.T, *apps.StatefulSet, invariantFunc)
 		obj func() *apps.StatefulSet
 	}{
-		{CreatesPods, simpleSetFn},
+		{CreatesPodsWithPodIndexLabelFeature, simpleSetFn},
+		{CreatesPodsWithoutPodIndexLabelFeature, simpleSetFn},
 		{ScalesUp, simpleSetFn},
 		{ScalesDown, simpleSetFn},
 		{ReplacesPods, largeSetFn},
@@ -208,7 +210,20 @@ func TestStatefulSetControl(t *testing.T) {
 	}
 }
 
-func CreatesPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+func CreatesPodsWithPodIndexLabelFeature(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+	createPods(t, set, invariants, true)
+}
+
+func CreatesPodsWithoutPodIndexLabelFeature(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+	createPods(t, set, invariants, false)
+}
+
+func createPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc, isPodIndexLabelEnabled bool) {
+	if !isPodIndexLabelEnabled {
+		// TODO: this will be removed in 1.35
+		featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, utilversion.MustParse("1.31"))
+	}
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodIndexLabel, isPodIndexLabelEnabled)
 	client := fake.NewSimpleClientset(set)
 	om, _, ssc := setupController(client)
 
@@ -229,20 +244,21 @@ func CreatesPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) 
 	if set.Status.UpdatedReplicas != 3 {
 		t.Error("Failed to set UpdatedReplicas correctly")
 	}
+
 	// Check all pods have correct pod index label.
-	if utilfeature.DefaultFeatureGate.Enabled(features.PodIndexLabel) {
-		selector, err := metav1.LabelSelectorAsSelector(set.Spec.Selector)
-		if err != nil {
-			t.Error(err)
-		}
-		pods, err := om.podsLister.Pods(set.Namespace).List(selector)
-		if err != nil {
-			t.Error(err)
-		}
-		if len(pods) != 3 {
-			t.Errorf("Expected 3 pods, got %d", len(pods))
-		}
-		for _, pod := range pods {
+	selector, err := metav1.LabelSelectorAsSelector(set.Spec.Selector)
+	if err != nil {
+		t.Error(err)
+	}
+	pods, err := om.podsLister.Pods(set.Namespace).List(selector)
+	if err != nil {
+		t.Error(err)
+	}
+	if len(pods) != 3 {
+		t.Errorf("Expected 3 pods, got %d", len(pods))
+	}
+	for _, pod := range pods {
+		if isPodIndexLabelEnabled {
 			podIndexFromLabel, exists := pod.Labels[apps.PodIndexLabel]
 			if !exists {
 				t.Errorf("Missing pod index label: %s", apps.PodIndexLabel)
@@ -251,6 +267,12 @@ func CreatesPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) 
 			podIndexFromName := strconv.Itoa(getOrdinal(pod))
 			if podIndexFromLabel != podIndexFromName {
 				t.Errorf("Pod index label value (%s) does not match pod index in pod name (%s)", podIndexFromLabel, podIndexFromName)
+			}
+		} else {
+			_, exists := pod.Labels[apps.PodIndexLabel]
+			if exists {
+				t.Errorf("Pod index label should not exist when feature gate is disabled: %s", apps.PodIndexLabel)
+				continue
 			}
 		}
 	}

